pkg/testing/integration/program.go

```diff
@@ -2199,12 +2170,7 @@ func (pt *ProgramTester) preparePythonProject(projinfo *engine.Projinfo) error {
 	} else {
 		venvPath := "venv"
 		if pt.opts.GetUseSharedVirtualEnv() {
-			requirementsPath := filepath.Join(cwd, "requirements.txt")
-			requirementsmd5, err := hashFile(requirementsPath)
-			if err != nil {
-				return err
-			}
-			pt.opts.virtualEnvDir = fmt.Sprintf("pulumi-venv-%x", requirementsmd5)
```

Comment:
It seems that there's some sort of interaction between different tests reading and writing to the same virtualenv.

Reply:
https://github.com/pulumi/pulumi/pull/14767